src/OSPSuite.Core/Domain/Services/QuantityValuesUpdater.cs

```diff
@@ -138,14 +141,23 @@ private IParameter getParameter(ModelConfiguration modelConfiguration, PathAndVa
          }
 
          //If the value is defined, this will be used instead of the formula (even if set previously)
-         if (parameterValue.Value.IsValid())
+         if (!parameterValue.Value.IsValid())
+            return;
+
+         var actualParameterValue = parameterValue.Value.Value;
+         if (parameter.Formula is ConstantFormula constantFormula)
          {
-            var actualParameterValue = parameterValue.Value.Value;
-            if (parameter.Formula is ConstantFormula constantFormula)
-               constantFormula.Value = actualParameterValue;
-            else
-               parameter.Value = actualParameterValue;
+            constantFormula.Value = actualParameterValue;
+            return;
          }
+
+         //now we have a non constant formula. Let's try to see if the reference can be resolved. If yes, we will simply set the value of the parameter
+         //Otherwise, we will create a new constant formula with the value
+
+         if (_formulaValidator.IsFormulaValid(parameter))
+            parameter.Value = actualParameterValue;
+         else
+            parameter.Formula = _formulaFactory.ConstantFormula(actualParameterValue, parameter.Dimension);
```

Comment:
@Yuri05 @PavelBal  This was this flag we had before
https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/blob/main/src/OSPSuite.Core/Domain/Services/QuantityValuesUpdater.cs#L70C1-L71C1
We used to always overwrite . Now we just set the value but of course does not work if the formula is not valid. It works now with this

so the order of call was the same as before. However, the logic had changed